### ISSUE TITLE
Link multiple devices or clients belonging to one person together (MULTIDEVICE)

### ISSUE THREAD

**Issue by drequivalent** [TokTok/c-toxcore#466]
In a real-world environment each user may have multiple computing and communication devices. For example, your machine at work, your phone and your home workstation. As Tox aims to be the means of user-friendly secure and reliable communications for everyone forever, we need to address the issues the users of multiple communication devices have with current Tox architecture.

As of now, if you and your recipient have multiple devices, and want to have a reliable commuication you have to:
 - manually add all each others' devices Tox IDs to all and each of your Tox clients.
 - somehow guess what device is your partner using when you want to send a message (statuses are unreliable by design)
 - if you want to make sure that your message gets through, you have to send your message to all known clients of your partner

This renders Tox virtually unusable in environments such as described earlier.

Thus, if Tox aims for widespread adoption, Tox must adopt a way to use multiple devices with the same visible identity. The architecture suggests this must be done via linking multiple independent devices to make a kind of a network of trust within one person's domain.

The linked devices must do the following main things:
 - synchronize contact lists
 - synchronize message history and incoming messages
 - synchronize calls (if one device gets a call, the others get a call too, the first one to pick up gets connected)
 - uniformly get offline messages no matter what device original message is sent to
 - synchronize file transfer (may be like calls, on first-come-first-served basis)
 - (optionally) allow call management as described in issue #92 

Linked devices may also be able to transfer messages to each other. It may not be useful for instant messgaing clients per se, but can be useful for Tox-based services and daemons.

**Comment by anoadragon453** [comment]
Additionally after syncing a new device, sending all previous message history is not necessarily required.

**Comment by drequivalent** [comment]
Any news on the matter?

**Comment by endoffile78** [comment]
@drequivalent @GrayHatter is working on multi-device.

**Comment by GrayHatter** [comment]
It's unfortunately slow going, I've since started a new job, and I'm also working on starting a 2nd business, so multi-device is lagging in development. But I did recently merge the new toktok api and it should generally work now. (no av, or file transfer support yet) https://github.com/uTox/toxcore/

**Comment by Nadahar** [comment]
I'm currently using Tox (qTox mostly), and for me the lack of multi-device support is the biggest hurdle (combined with the fact that the timestamps are from when a message is received, not sent). Having to use multiple accounts makes it very hard to communicate because you never know what has been sent to your "other" account.

**Comment by volth** [comment]
It seems to be heavily tied to the offline messaging proposal: it is a real use case when two devices belonging to a single person never go online together. If a direct connect is required to sync, then they cannot sync.

**Comment by Sami32** [comment]
Definitively a thing to have to avoid copy / paste and madness.
Trying to find what is the device used by a developper at a given time make me crazy !

**Comment by Nadahar** [comment]
I agree that an offline messages implementation is needed to get a "proper" multi-device solution, but in the meanwhile I would prefer having multi-device support with synchronization only if both clients were online (or without synchronization at all in worst case) than having to use separate accounts.

**Comment by WojAbuk** [comment]
Can I somehow help speed up my multi-device deployment? I know the basics of C / C ++ and some other languages.

**Comment by zoff99** [comment]
@WojAbuk please either join us on IRC or via Matrix:

#toktok on freenode

https://matrix.to/#/#freenode_#toktok:matrix.org

**Comment by lemmy04** [comment]
...any news on this? without multi device support I would have ELEVEN separate tox IDs...

**Comment by zoff99** [comment]
@lemmy04 we need people to help out with knowledge in C coding, then we can speed things up
